### PR TITLE
chore: upgrade PHPUnit 9 => 10

### PR DESCRIPTION
**Pull request by Chris53897**
changes for sebastian/comparator

https://github.com/moneyphp/money/pull/746 rebased against master.
Migrate Config PHPUnit 9 => 10
Only Allow PHPUnit 10.

I did not have a closer look to the comperator changes.

### CODE REVIEW

**Comment by Chris53897** [comment]
Tests are failing, because of
DeprecatedMethod: The method PHPUnit\Framework\TestCase::setLocale has been marked as deprecated (see https://psalm.dev/001)

**Comment by Chris53897** [comment]
I replaced it with direct calls to `setlocale()`.
But i am not sure if this is enough.

File: tests/Calculator/LocaleAwareBcMathCalculatorTest.php [hunk 1]
```diff
@@ -13,6 +15,6 @@ public function setUp(): void
     {
         parent::setUp();
 
-        $this->setLocale(LC_ALL, 'ru_RU.UTF-8');
+        setlocale(LC_ALL, 'ru_RU.UTF-8');
```
Comment (frederikbosch) [inline tests/Calculator/LocaleAwareBcMathCalculatorTest.php:18]: I would add a file called `tests/LocaleTest.php`, with the following content.

```php
declare(strict_types=1);

namespace Tests\Money;

trait LocaleTest
{
    public static function setLocale(int $category, string $locale, \Closure $callback)
    {
        $currentLocale= \setlocale($category, 0);
        try {
            \setlocale($category, $locale);
            $callback();
        } finally {
            \setlocale($category, $currentLocale);
        }
    }
}
```

And then remove `setlocale` from `setUp` but simply use it in the test file as follows

```php
use Tests\Money\LocaleTest;

public static function the_test()
{
    LocaleTest::setLocale(LC_ALL, 'ru_RU.UTF-8', function () {
        // test content here
    });
}

```

**Review by frederikbosch — COMMENTED** [review]
This looks good. Only with regard to `setlocale` I would solve that differently

File: tests/Calculator/LocaleAwareBcMathCalculatorTest.php (hunk 1 above)
Comment (Chris53897) [reply tests/Calculator/LocaleAwareBcMathCalculatorTest.php:18]: I followed your suggestion. But i did need to make some changes to make it work.
The name setLocale is already taken with other parameters.

Filename changed from LocaleTest to Locale

I am not sure if i did everythink correct. Please have a look.

**Pull request closed without merging**